common/tree/script_test.go

```diff
@@ -624,3 +628,366 @@ func TestCLTVMultisigClosure(t *testing.T) {
+				txscript.OP_15,     // stack: b a isSize16 size(a) 15
+				txscript.OP_EQUAL,  // stack: a b isSize16 isSize15
+				txscript.OP_BOOLOR, // stack: a b isValidSize
+				txscript.OP_NOTIF,
+				txscript.OP_2DROP,
+				txscript.OP_0, // stack : 0
+				txscript.OP_ELSE,
+				txscript.OP_SIZE, // stack: b a size(a)
+				txscript.OP_SWAP, // stack: b size(a) a
+				txscript.OP_DROP, // stack: b size(a)
+				txscript.OP_SWAP, // stack: size(a) b
+				txscript.OP_SIZE, // stack: size(a) b size(b)
+				txscript.OP_SWAP, // stack: size(a) size(b) b
+				txscript.OP_DROP, // stack: size(a) size(b)
+				txscript.OP_EQUAL,
+				txscript.OP_ENDIF,
+				txscript.OP_ENDIF,
+			}...),
+			witness:     wire.TxWitness{rand1, rand2},
+			returnValue: false,
+			expectErr:   false,
+		},
+		{
+			name: "valid coinflip same lengths",
+			script: append(append(append(append([]byte{
+				txscript.OP_2DUP,    // stack: a b a b
+				txscript.OP_SHA256,  // stack: a b a h(b)
+				txscript.OP_DATA_32, // stack: a b a h(b) hash2
+			}, hash1[:]...), []byte{
+				txscript.OP_EQUALVERIFY, // stack: a b a
+				txscript.OP_SHA256,      // stack: a b h(a)
+				txscript.OP_DATA_32,     // stack: a b h(a) hash1
+			}...), hash1[:]...), []byte{
+				txscript.OP_EQUALVERIFY, // stack: a b
+				txscript.OP_SIZE,        // stack: a b size(b)
+				txscript.OP_DUP,         // stack: a b size(b) size(b)
+				txscript.OP_16,          // stack: a b size(b) size(b) 16
+				txscript.OP_EQUAL,       // stack: a b size(b) isSize16
+				txscript.OP_SWAP,        // stack: a b isSize16 size(b)
+				txscript.OP_15,          // stack: a b isSize16 size(b) 15
+				txscript.OP_EQUAL,       // stack: a b isSize16 isSize15
+				txscript.OP_BOOLOR,      // stack: a b isValidSize
+				txscript.OP_NOTIF,
+				txscript.OP_2DROP,
+				txscript.OP_1, // stack : 1
+				txscript.OP_ELSE,
+				txscript.OP_SWAP,   // stack: b a
+				txscript.OP_SIZE,   // stack: b a size(a)
+				txscript.OP_DUP,    // stack: b a size(a) size(a)
+				txscript.OP_16,     // stack: b a size(a) size(a) 16
+				txscript.OP_EQUAL,  // stack: b a size(a) isSize16
+				txscript.OP_SWAP,   // stack: b a isSize16 size(a)
+				txscript.OP_15,     // stack: b a isSize16 size(a) 15
+				txscript.OP_EQUAL,  // stack: a b isSize16 isSize15
+				txscript.OP_BOOLOR, // stack: a b isValidSize
+				txscript.OP_NOTIF,
+				txscript.OP_2DROP,
+				txscript.OP_0, // stack : 0
+				txscript.OP_ELSE,
+				txscript.OP_SIZE, // stack: b a size(a)
+				txscript.OP_SWAP, // stack: b size(a) a
+				txscript.OP_DROP, // stack: b size(a)
+				txscript.OP_SWAP, // stack: size(a) b
+				txscript.OP_SIZE, // stack: size(a) b size(b)
+				txscript.OP_SWAP, // stack: size(a) size(b) b
+				txscript.OP_DROP, // stack: size(a) size(b)
+				txscript.OP_EQUAL,
+				txscript.OP_ENDIF,
+				txscript.OP_ENDIF,
+			}...),
+			witness:     wire.TxWitness{rand1, rand1},
+			returnValue: true,
+			expectErr:   false,
+		},
+		{
+			name: "invalid first element length",
```

Comment:
can we make this test simpler? we want to "test invalid first elem length", let's do it with the simplest possible script, not this nightmare of script